### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by ggrig**
97 tests testing all the controllers

image

### CODE REVIEW

File: Source/Service/Controllers/RebuildController.cs [hunk 1]
```diff
@@ -360,10 +360,16 @@ public async Task<IActionResult> RebuildFromFormProtectedFile([FromForm][Require
...
+                    // The following fails to create a file
+                    //
+                    //using (Stream fileStream = new FileStream(protectedZipFilePath, FileMode.Create))
+                    //{
+                    //    await file.CopyToAsync(fileStream);
+                    //}
+                    //
+                    // Replacing it with WriteAllBytes
+
+                    System.IO.File.WriteAllBytes(protectedZipFilePath, fileBytes);
```
Comment (mohitsengarr) [inline Source/Service/Controllers/RebuildController.cs:372]: please use the Async version of it. WriteAllBytesAsync

File: Source/Service/Controllers/RebuildController.cs [hunk 2]
```diff
@@ -360,10 +360,16 @@ public async Task<IActionResult> RebuildFromFormProtectedFile([FromForm][Require
                         Directory.CreateDirectory(tempFolderPath);
                     }
 
-                    using (Stream fileStream = new FileStream(protectedZipFilePath, FileMode.Create))
-                    {
-                        await file.CopyToAsync(fileStream);
-                    }
+                    // The following fails to create a file
```
Comment (mohitsengarr) [inline Source/Service/Controllers/RebuildController.cs:363]: I don't understand why it is failing for you as it is already working. However, can you please remove the commented code.

**Review by mohitsengarr — CHANGES_REQUESTED** [review]

File: Source/Service/Controllers/RebuildController.cs (hunk 2 above)
Comment (ggrig) [reply Source/Service/Controllers/RebuildController.cs:363]: I also do not have a good explanation. It seems this relates to the utilization of a Mock object for  IFormFile. The observation is that once created and read it does not allow a second read.

File: Source/Service/Controllers/RebuildController.cs (hunk 1 above)
Comment (ggrig) [reply Source/Service/Controllers/RebuildController.cs]: Done. Thanks for pointing that out

**Review by mohitsengarr — APPROVED** [review]

**Pull request merged**